src/common/hooks/useSticky.tsx

```diff
@@ -1,40 +1,28 @@
-import { useEffect, useState } from 'react';
-import { useLocation } from 'react-router-dom';
-import { authRoutes } from '../../routes';
+import { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import {
+  setIsNavSticky,
+  setYOffset,
+  updateYOffset,
+} from '../../store/app/actions';
 
-export interface ISticky {
-  sticky: boolean;
-  home: boolean;
-  locationUpdate: () => void;
-}
+import { useScrollHandler } from './useScrollHandler';
 
-function useSticky(): ISticky {
-  const [sticky, setSticky] = useState(false);
-  const [home, setHome] = useState(false);
-  const location = useLocation();
+export const useSticky = (): void => {
+  const dispatch = useDispatch();
```

Comment:
I moved the state into a global store since it is being used at multiple levels of the app. This allowed me to let useSticky cleanup its event listeners.